### PR TITLE
qremotecontrol-server: 2.4.1 -> unstable-2014-11-05, use Qt5 

### PR DESCRIPTION
**Pull request by fgaz**
###### Description of changes

Qt4 is going to be removed: https://github.com/NixOS/nixpkgs/pull/174634

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/misc/qremotecontrol-server/default.nix [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/qrc/gitcode";
+    rev = "8f1c55eac10ac8af974c3c20157d90ef57f7308a";
+    sha256 = "sha256-AfFScec5/emG/f+yc5Zn37USIEWzGP/sBifE6Kx8d0E=";
   };
 
-  nativeBuildInputs = [ qmake4Hook ];
-  buildInputs = [ qt4 xorg.libXtst ];
+  patches = [
+    ./0001-Another-Qt5-fix.patch
```
Comment (veprbl) [inline pkgs/servers/misc/qremotecontrol-server/default.nix:21]: Missing file

Comment (fgaz) [reply pkgs/servers/misc/qremotecontrol-server/default.nix:21]: oops, added

**Review by klemensn — CHANGES_REQUESTED** [review]
`meta.homepage` exists but has nothing to do with this program.
`meta.downloadPage` yields "File not found".

https://showroom.qt.io/qremotecontrol/ looks official.
This website still links to http://qremote.org, but I'm certain they're just outdated.

qremote.org's WHOIS data shows
```
Updated Date: 2022-06-14T01:12:56Z
Creation Date: 2022-03-17T14:30:15Z
```

File: pkgs/servers/misc/qremotecontrol-server/default.nix (hunk 1 above)
Comment (fgaz) [reply pkgs/servers/misc/qremotecontrol-server/default.nix]: done

**Comment by fgaz** [comment]
I'd rather link to he sourceforge page then. done.

**Review by klemensn — APPROVED** [review]

File: pkgs/servers/misc/qremotecontrol-server/default.nix [hunk 2]
```diff
@@ -1,21 +1,35 @@
...
+, mkDerivation
+, fetchgit
+, qmake
+, wrapQtAppsHook
+, qtbase
 , xorg
 }:
 
-stdenv.mkDerivation rec {
+mkDerivation rec {
```
Comment (veprbl) [inline pkgs/servers/misc/qremotecontrol-server/default.nix:10]:
```suggestion
stdenv.mkDerivation rec {
```
`libsForQt5.mkDerivation` is not composable. What it used to do is now done in wrapQtAppsHook.

Comment (fgaz) [reply pkgs/servers/misc/qremotecontrol-server/default.nix]: Fixed. When did that change?

**Comment by fgaz** [comment]
Ping

File: pkgs/servers/misc/qremotecontrol-server/default.nix [hunk 3]
```diff
@@ -1,21 +1,35 @@
...
-  version = "2.4.1";
+  version = "unstable-2014-11-05"; # basically 2.4.2 + qt5
 
-  src = fetchurl {
-    url = "mirror://sourceforge/project/qrc/${version}/qremotecontrol-${version}.tar.bz2";
-    sha256 = "07hzc9959a56b49jgmcv8ry8b9sppklvqs9kns3qjj3v9d22nbrp";
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/qrc/gitcode";
+    rev = "8f1c55eac10ac8af974c3c20157d90ef57f7308a";
+    sha256 = "sha256-AfFScec5/emG/f+yc5Zn37USIEWzGP/sBifE6Kx8d0E=";
```
Comment (ajs124) [inline pkgs/servers/misc/qremotecontrol-server/default.nix:17]: I get
```
error: hash mismatch in fixed-output derivation '/nix/store/0c2dp4khkq58hpy350f4q2qqiaqmpznb-gitcode-8f1c55e.drv':
         specified: sha256-AfFScec5/emG/f+yc5Zn37USIEWzGP/sBifE6Kx8d0E=
            got:    sha256-fdUiuW5mdXWewvrJ2LKuEnB2qrPPALSg/cjOEBnc5YY=
```
:thinking:

Comment (ajs124) [reply pkgs/servers/misc/qremotecontrol-server/default.nix]: weird… suggests that the hash isn't stable, which it really shouldn't be.

File: pkgs/servers/misc/qremotecontrol-server/default.nix (hunk 2 above)
Comment (veprbl) [reply pkgs/servers/misc/qremotecontrol-server/default.nix]: Since #108888

File: pkgs/servers/misc/qremotecontrol-server/default.nix (hunk 3 above)
Comment (fgaz) [reply pkgs/servers/misc/qremotecontrol-server/default.nix]: I'm still getting the same hash...

**Comment by fgaz** [comment]
@GrahamcOfBorg build qremotecontrol-server

File: pkgs/servers/misc/qremotecontrol-server/default.nix (hunk 3 above)
Comment (fgaz) [reply pkgs/servers/misc/qremotecontrol-server/default.nix:17]: I rebased. That might do the trick, like in #179560

**Pull request merged**